compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
I am not sure I understand your question completely. So I am proposing to keep setVP as the following:

void setVP(OMR::ValuePropagation *vp) {_vp = vp;}

And when we invoke setVP, we need to make sure the right VP object (OMR or VP for another project) is passed as the argument to setVP. After that, invoking virtual methods on the right VP object will dispatch to the desired methods automatically.